### PR TITLE
[coap] remove otCoapMessage from CoapSecure::DefaultHandler()

### PR DESCRIPTION
**Pull request by ozanoner**

### CODE REVIEW

**Comment by ozanoner** [comment]
I signed it!

**Comment by bukepo** [comment]
Could you please add `CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER` to https://github.com/openthread/openthread/blob/master/examples/platforms/posix/openthread-core-posix-config.h?

**Comment by bukepo** [comment]
Any update?

**Comment by ozanoner** [comment]
Hi Yakun, sorry for the late response. I will update it today.

**Comment by ozanoner** [comment]
> Could you please add `CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER` to https://github.com/openthread/openthread/blob/master/examples/platforms/posix/openthread-core-posix-config.h?

@bukepo Travis-CI gives redefinition error. Are you sure about above addition? 
CI output:
```
/home/travis/build/ozanoner/openthread/examples/../src/cli/cli_coap_secure.hpp:47:0: error: "CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER" redefined [-Werror]
 #define CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER 0
```

**Comment by jwhui** [comment]
@ozanoner, we should remove the `#define CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER 0` from `src/cil/cli_coap_secure.hpp`.

File: src/cli/cli_coap_secure.cpp [hunk 1]
```diff
@@ -586,30 +586,26 @@ void CoapSecure::HandleClientResponse(otMessage *aMessage, const otMessageInfo *
 }
 
 #if CLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER
-void OTCALL CoapSecure::DefaultHandle(void *               aContext,
-                                      otCoapMessage *      aHeader,
-                                      otMessage *          aMessage,
-                                      const otMessageInfo *aMessageInfo)
+void OTCALL CoapSecure::DefaultHandle(void *aContext, otMessage *aMessage, const otMessageInfo *aMessageInfo)
```
Comment (jwhui) [inline src/cli/cli_coap_secure.cpp]: nit: let's rename this to `CoapSecure::DefaultHandler()` to be consistent with the rest of OpenThread.

**Comment by jwhui** [comment]
@ozanoner, the Google CLA check is not passing. Can you review the comment from @googlebot  above? Thanks.

**Comment by ozanoner** [comment]
> @ozanoner, the Google CLA check is not passing. Can you review the comment from @googlebot above? Thanks.

Hi Jonathan, I had already signed the CLA before, so I thought you would do something else about it. I double-checked now, and it says 'you already signed'. I updated it again. Do I need to do something different?

**Comment by jwhui** [comment]
@ozanoner, just to double check - are you using the same email in your git commits as the Google CLA?

**Comment by ozanoner** [comment]
> @ozanoner, just to double check - are you using the same email in your git commits as the Google CLA?

Yes. I checked the CLA contact info once more. I confirm that, Signed date: Mar 14, email address: my email address, name: my name, github username: ozanoner as we use here, and for commits. I don't see any wrong or incompatible info

**Comment by ozanoner** [comment]
I signed it!

**Comment by jwhui** [comment]
@ozanoner, it seems your commit history is using two different email addresses for various commits. Can you update all of them to be consistent?

**Comment by ozanoner** [comment]
> @ozanoner, it seems your commit history is using two different email addresses for various commits. Can you update all of them to be consistent?

Updated!

**Comment by jwhui** [comment]
@ozanoner, I still see commits using two different email addresses. Can you check again?

For example:
```
commit 54b462a9d171101526cc350d0bd208982553f56d
Merge: 94933a4f 7a34b9d8
Author: Vedat Ozan Oner <ozan.oner@gmail.com>
Date:   Thu Mar 28 10:06:27 2019 +0000
```
and
```
commit b81b977a4cd21e74c8f5d92a2984e5ec89958219
Author: Vedat Ozan Oner <ozan.oner@mevoo.co.uk>
Date:   Wed Mar 13 17:37:40 2019 +0000
```

**Comment by ozanoner** [comment]
Hi Jonathan, maybe it would be easier to go with a fresh fork. git gives many errors after updating history, and rejects push. when I pull, then history comes back and my wrong email address reappears. do you have any other suggestion? sorry for the inconvenience.

**Comment by jwhui** [comment]
It's probably best if you start with a clean branch. In the future, you should probably use `git pull --rebase` to ensure your commits stay on top.

**Comment by jwhui** [comment]
Replaced by #3737.

**Pull request closed without merging**